[Cocoapods] Support compiling plain C

I tried to build Sentry pod, but I got compiler's error regarding C sources
```
clang++ -c -arch arm64 -target arm-apple-darwin19 -m64 -O2 -g -stdlib=libc++ -miphoneos-version-min=11.0 -isysroot <redacted>/platformsdk/iPhoneOS16.2.sdk -nostdinc++ -isystem <redacted>/platformsdk/iPhoneOS16.2.sdk/usr/include/c++/v1  -DDLIB_LOG_DOMAIN="SENTRY" -DDDF_EXPOSE_DESCRIPTORS -DDM_PLATFORM_IOS -DDM_DEBUG  -fno-exceptions -fvisibility=hidden -fobjc-arc -Iupload/ -ICocoaPodsService/Pods/Sentry/  -F/var/folders/nq/zlqy380j2fb1xwwkry71j9jw0000gn/T/job11030230481155200560/CocoaPodsService/frameworks/lib/arm64-ios  -I<redacted>/sdk/b1f3977fbad9230bbf048e67e779d1e3c86f2cf6/defoldsdk//include -I<redacted>/sdk/b1f3977fbad9230bbf048e67e779d1e3c86f2cf6/defoldsdk//sdk/include -I<redacted>/sdk/b1f3977fbad9230bbf048e67e779d1e3c86f2cf6/defoldsdk//ext/include   CocoaPodsService/Pods/Sentry/Sources/SentryCrash/Recording/Tools/SentryCrashStackCursor_MachineContext.c -obuild/SentryCrashStackCursor_MachineContext.c_3.o
clang: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]
In file included from CocoaPodsService/Pods/Sentry/Sources/SentryCrash/Recording/Tools/SentryCrashStackCursor_MachineContext.c:28:
CocoaPodsService/Pods/Sentry/Sources/SentryCrash/Recording/Tools/SentryCrashMemory.h:69:26: error: expected ')'
    const void *restrict const src, void *restrict const dst, int byteCount);
```
As I understood `restrict` keyword used only in C.
Podfile:
```
platform :ios, '11.0'

pod 'Sentry', :git => 'https://github.com/getsentry/sentry-cocoa.git', :tag => '7.31.5'
```

--- a/server/src/main/java/com/defold/extender/Extender.java
+++ b/server/src/main/java/com/defold/extender/Extender.java
@@ -680,28 +680,7 @@ class Extender {
         List<String> objs = new ArrayList<>();
         List<String> commands = new ArrayList<>();
 
-        // build lists of flags and defines per supported language
-        List<String> defines = new ArrayList<>(pod.defines);
-        List<String> flagsC = new ArrayList<>(pod.flags);
-        List<String> flagsCpp = new ArrayList<>(pod.flags);
-        List<String> flagsObjC = new ArrayList<>(pod.flags);
-        List<String> flagsObjCpp = new ArrayList<>(pod.flags);
-        if (platform.contains("ios")) {
-            defines.addAll(pod.ios_defines);
-            flagsC.addAll(pod.ios_flags);
-            flagsCpp.addAll(pod.ios_flags);
-            flagsObjC.addAll(pod.ios_flags);
-            flagsObjCpp.addAll(pod.ios_flags);
-        }
-        else if (platform.contains("osx")) {
-            defines.addAll(pod.osx_defines);
-            flagsC.addAll(pod.osx_flags);
-            flagsCpp.addAll(pod.osx_flags);
-            flagsObjC.addAll(pod.osx_flags);
-            flagsObjCpp.addAll(pod.osx_flags);
-        }
-
-        // TEMP FIX! remove hardcoded flags from context (should be removed from build.yaml perhaps?)
+        // remove hardcoded flags from context
         Map<String, Object> trimmedContext = ExtenderUtil.mergeContexts(manifestContext, new HashMap<>());
         removeFlag("-fno-exceptions", (List<String>)trimmedContext.get("flags"));
         removeFlag("-stdlib=libc++", (List<String>)trimmedContext.get("flags"));
@@ -712,47 +691,33 @@ class Extender {
         Map<String, Object> podContextCpp = new HashMap<>();
         Map<String, Object> podContextObjC = new HashMap<>();
         Map<String, Object> podContextObjCpp = new HashMap<>();
-        podContextC.put("flags", flagsC);
-        podContextC.put("defines", defines);
-        podContextCpp.put("flags", flagsCpp);
-        podContextCpp.put("defines", defines);
-        podContextObjC.put("flags", flagsObjC);
-        podContextObjC.put("defines", defines);
-        podContextObjCpp.put("flags", flagsObjCpp);
-        podContextObjCpp.put("defines", defines);
+
+        if (platform.contains("ios")) {
+            podContextC.put("flags", new ArrayList<String>(pod.flags.ios.c));
+            podContextC.put("defines", new ArrayList<String>(pod.defines.ios));
+            podContextCpp.put("flags", new ArrayList<String>(pod.flags.ios.cpp));
+            podContextCpp.put("defines", new ArrayList<String>(pod.defines.ios));
+            podContextObjC.put("flags", new ArrayList<String>(pod.flags.ios.objc));
+            podContextObjC.put("defines", new ArrayList<String>(pod.defines.ios));
+            podContextObjCpp.put("flags", new ArrayList<String>(pod.flags.ios.objcpp));
+            podContextObjCpp.put("defines", new ArrayList<String>(pod.defines.ios));
+        }
+        else if (platform.contains("osx")) {
+            podContextC.put("flags", new ArrayList<String>(pod.flags.osx.c));
+            podContextC.put("defines", new ArrayList<String>(pod.defines.osx));
+            podContextCpp.put("flags", new ArrayList<String>(pod.flags.osx.cpp));
+            podContextCpp.put("defines", new ArrayList<String>(pod.defines.osx));
+            podContextObjC.put("flags", new ArrayList<String>(pod.flags.osx.objc));
+            podContextObjC.put("defines", new ArrayList<String>(pod.defines.osx));
+            podContextObjCpp.put("flags", new ArrayList<String>(pod.flags.osx.objcpp));
+            podContextObjCpp.put("defines", new ArrayList<String>(pod.defines.osx));
+        }
+
+        // get the final contexts per supported language
         Map<String, Object> mergedContextWithPodsForC = ExtenderUtil.mergeContexts(trimmedContext, podContextC);
         Map<String, Object> mergedContextWithPodsForCpp = ExtenderUtil.mergeContexts(trimmedContext, podContextCpp);
         Map<String, Object> mergedContextWithPodsForObjC = ExtenderUtil.mergeContexts(trimmedContext, podContextObjC);
         Map<String, Object> mergedContextWithPodsForObjCpp = ExtenderUtil.mergeContexts(trimmedContext, podContextObjCpp);
-
-        // get the flags per supported language and do some fix-up
-        List<String> mergedCFlags = (List<String>)mergedContextWithPodsForC.get("flags");
-        List<String> mergedCppFlags = (List<String>)mergedContextWithPodsForCpp.get("flags");
-        List<String> mergedObjCFlags = (List<String>)mergedContextWithPodsForObjC.get("flags");
-        List<String> mergedObjCppFlags = (List<String>)mergedContextWithPodsForObjCpp.get("flags");
-        // fix c flags
-        removeFlag("-std=c++", mergedCFlags);
-        removeFlag("-stdlib=", mergedCFlags);
-        removeFlag("-fexceptions", mergedCFlags);
-        removeFlag("-fno-exceptions", mergedCFlags);
-        removeFlag("-fcxx-exceptions", mergedCFlags);
-        removeFlag("-fno-cxx-exceptions", mergedCFlags);
-        mergedCFlags.add("--language=c");
-
-        // fix cpp flags
-        removeFlag("-std=c99", mergedCppFlags);
-        mergedCppFlags.add("--language=c++");
-        
-        // fix objc flags
-        removeFlag("-std=c99", mergedObjCFlags);
-        removeFlag("-std=c++", mergedObjCFlags);
-        removeFlag("-stdlib=", mergedObjCFlags);
-        mergedObjCFlags.add("--language=objective-c");
-
-        // fix objcpp flags
-        removeFlag("-std=c99", mergedCppFlags);
-        mergedObjCppFlags.add("--language=objective-c++");
-
 
         for (File src : pod.sourceFiles) {
             String extension = FilenameUtils.getExtension(src.getAbsolutePath());

--- a/server/src/main/java/com/defold/extender/Extender.java
+++ b/server/src/main/java/com/defold/extender/Extender.java
@@ -680,39 +680,79 @@ class Extender {
         List<String> objs = new ArrayList<>();
         List<String> commands = new ArrayList<>();
 
-        // per supported language build lists of flags and defines
+        // build lists of flags and defines per supported language
         List<String> defines = new ArrayList<>(pod.defines);
+        List<String> flagsC = new ArrayList<>(pod.flags);
         List<String> flagsCpp = new ArrayList<>(pod.flags);
         List<String> flagsObjC = new ArrayList<>(pod.flags);
-        List<String> flagsC = new ArrayList<>(pod.flags);
+        List<String> flagsObjCpp = new ArrayList<>(pod.flags);
         if (platform.contains("ios")) {
+            defines.addAll(pod.ios_defines);
+            flagsC.addAll(pod.ios_flags);
             flagsCpp.addAll(pod.ios_flags);
             flagsObjC.addAll(pod.ios_flags);
-            flagsC.addAll(pod.ios_flags);
-            defines.addAll(pod.ios_defines);
+            flagsObjCpp.addAll(pod.ios_flags);
         }
         else if (platform.contains("osx")) {
+            defines.addAll(pod.osx_defines);
+            flagsC.addAll(pod.osx_flags);
             flagsCpp.addAll(pod.osx_flags);
             flagsObjC.addAll(pod.osx_flags);
-            flagsC.addAll(pod.osx_flags);
-            defines.addAll(pod.osx_defines);
+            flagsObjCpp.addAll(pod.osx_flags);
         }
-        removeFlag("-std=c++", flagsObjC);
-        removeFlag("-std=c++", flagsC);
+
+        // TEMP FIX! remove hardcoded flags from context (should be removed from build.yaml perhaps?)
+        Map<String, Object> trimmedContext = ExtenderUtil.mergeContexts(manifestContext, new HashMap<>());
+        removeFlag("-fno-exceptions", (List<String>)trimmedContext.get("flags"));
+        removeFlag("-stdlib=libc++", (List<String>)trimmedContext.get("flags"));
 
         // create contexts per supported language
+        // add pod flags and defines
+        Map<String, Object> podContextC = new HashMap<>();
         Map<String, Object> podContextCpp = new HashMap<>();
         Map<String, Object> podContextObjC = new HashMap<>();
-        Map<String, Object> podContextC = new HashMap<>();
+        Map<String, Object> podContextObjCpp = new HashMap<>();
+        podContextC.put("flags", flagsC);
+        podContextC.put("defines", defines);
         podContextCpp.put("flags", flagsCpp);
         podContextCpp.put("defines", defines);
         podContextObjC.put("flags", flagsObjC);
         podContextObjC.put("defines", defines);
-        podContextC.put("flags", flagsC);
-        podContextC.put("defines", defines);
-        Map mergedContextWithPodsForCpp = ExtenderUtil.mergeContexts(manifestContext, podContextCpp);
-        Map mergedContextWithPodsForObjC = ExtenderUtil.mergeContexts(manifestContext, podContextObjC);
-        Map mergedContextWithPodsForC = ExtenderUtil.mergeContexts(manifestContext, podContextC);
+        podContextObjCpp.put("flags", flagsObjCpp);
+        podContextObjCpp.put("defines", defines);
+        Map<String, Object> mergedContextWithPodsForC = ExtenderUtil.mergeContexts(trimmedContext, podContextC);
+        Map<String, Object> mergedContextWithPodsForCpp = ExtenderUtil.mergeContexts(trimmedContext, podContextCpp);
+        Map<String, Object> mergedContextWithPodsForObjC = ExtenderUtil.mergeContexts(trimmedContext, podContextObjC);
+        Map<String, Object> mergedContextWithPodsForObjCpp = ExtenderUtil.mergeContexts(trimmedContext, podContextObjCpp);
+
+        // get the flags per supported language and do some fix-up
+        List<String> mergedCFlags = (List<String>)mergedContextWithPodsForC.get("flags");
+        List<String> mergedCppFlags = (List<String>)mergedContextWithPodsForCpp.get("flags");
+        List<String> mergedObjCFlags = (List<String>)mergedContextWithPodsForObjC.get("flags");
+        List<String> mergedObjCppFlags = (List<String>)mergedContextWithPodsForObjCpp.get("flags");
+        // fix c flags
+        removeFlag("-std=c++", mergedCFlags);
+        removeFlag("-stdlib=", mergedCFlags);
+        removeFlag("-fexceptions", mergedCFlags);
+        removeFlag("-fno-exceptions", mergedCFlags);
+        removeFlag("-fcxx-exceptions", mergedCFlags);
+        removeFlag("-fno-cxx-exceptions", mergedCFlags);
+        mergedCFlags.add("--language=c");
+
+        // fix cpp flags
+        removeFlag("-std=c99", mergedCppFlags);
+        mergedCppFlags.add("--language=c++");
+        
+        // fix objc flags
+        removeFlag("-std=c99", mergedObjCFlags);
+        removeFlag("-std=c++", mergedObjCFlags);
+        removeFlag("-stdlib=", mergedObjCFlags);
+        mergedObjCFlags.add("--language=objective-c");
+
+        // fix objcpp flags
+        removeFlag("-std=c99", mergedCppFlags);
+        mergedObjCppFlags.add("--language=objective-c++");
+
 
         for (File src : pod.sourceFiles) {
             String extension = FilenameUtils.getExtension(src.getAbsolutePath());
@@ -720,19 +760,21 @@ class Extender {
                 throw new ExtenderException("Unable to build '" + pod.name + "' since it includes Swift source files");
             }
             else {
+                final int i = getAndIncreaseNameCount();
+                File o = null;
                 // use the correct context depending on the source file language
-                Map mergedContextWithPods = null;
                 if (extension.equals("c")) {
-                    mergedContextWithPods = mergedContextWithPodsForC;
+                    o = addCompileFileCppStatic(i, pod.dir, src, mergedContextWithPodsForC, commands);
                 }
-                else if (extension.equals("m") || extension.equals("mm")) {
-                    mergedContextWithPods = mergedContextWithPodsForObjC;
+                else if (extension.equals("m")) {
+                    o = addCompileFileCppStatic(i, pod.dir, src, mergedContextWithPodsForObjC, commands);
+                }
+                else if (extension.equals("mm")) {
+                    o = addCompileFileCppStatic(i, pod.dir, src, mergedContextWithPodsForObjCpp, commands);
                 }
                 else {
-                    mergedContextWithPods = mergedContextWithPodsForCpp;
+                    o = addCompileFileCppStatic(i, pod.dir, src, mergedContextWithPodsForCpp, commands);
                 }
-                final int i = getAndIncreaseNameCount();
-                File o = addCompileFileCppStatic(i, pod.dir, src, mergedContextWithPods, commands);
                 objs.add(ExtenderUtil.getRelativePath(jobDirectory, o));
             }
         }

--- a/server/src/main/java/com/defold/extender/Extender.java
+++ b/server/src/main/java/com/defold/extender/Extender.java
@@ -665,25 +665,14 @@ class Extender {
         return objs;
     }
 
-    // remove flags starting with a specific string
-    private void removeFlag(String flagToRemoveStartsWith, List<String> flags) {
-        for (int i = flags.size() - 1; i >= 0; i--) {
-            String flag = flags.get(i);
-            if (flag.startsWith(flagToRemoveStartsWith)) {
-                flags.remove(i);
-            }
-        }
-    }
-
     // compile the source files of a pod and return a list of object files
     private List<String> compilePodSourceFiles(PodSpec pod, Map<String, Object> manifestContext) throws IOException, InterruptedException, ExtenderException {
         List<String> objs = new ArrayList<>();
         List<String> commands = new ArrayList<>();
 
-        // remove hardcoded flags from context
+        // clean up flags from context
         Map<String, Object> trimmedContext = ExtenderUtil.mergeContexts(manifestContext, new HashMap<>());
-        removeFlag("-fno-exceptions", (List<String>)trimmedContext.get("flags"));
-        removeFlag("-stdlib=libc++", (List<String>)trimmedContext.get("flags"));
+        trimmedContext.put("flags", new ArrayList<String>());
 
         // create contexts per supported language
         // add pod flags and defines

--- a/server/src/main/java/com/defold/extender/services/CocoaPodsService.java
+++ b/server/src/main/java/com/defold/extender/services/CocoaPodsService.java
@@ -51,12 +51,11 @@ public class CocoaPodsService {
         public List<String> getAllPodLibs(String platform) {
             Set<String> libs = new HashSet<>();
             for (PodSpec pod : pods) {
-                libs.addAll(pod.libraries);
                 if (platform.contains("ios")) {
-                    libs.addAll(pod.ios_libraries);
+                    libs.addAll(pod.libraries.ios);
                 }
                 else if (platform.contains("osx")) {
-                    libs.addAll(pod.osx_libraries);
+                    libs.addAll(pod.libraries.osx);
                 }
             }
             return new ArrayList<String>(libs);
@@ -65,12 +64,11 @@ public class CocoaPodsService {
         public List<String> getAllPodLinkFlags(String platform) {
             Set<String> flags = new HashSet<>();
             for (PodSpec pod : pods) {
-                flags.addAll(pod.linkflags);
                 if (platform.contains("ios")) {
-                    flags.addAll(pod.ios_linkflags);
+                    flags.addAll(pod.linkflags.ios);
                 }
                 else if (platform.contains("osx")) {
-                    flags.addAll(pod.osx_linkflags);
+                    flags.addAll(pod.linkflags.osx);
                 }
             }
             return new ArrayList<String>(flags);
@@ -79,12 +77,11 @@ public class CocoaPodsService {
         public List<String> getAllPodFrameworks(String platform) {
             Set<String> frameworks = new HashSet<>();
             for (PodSpec pod : pods) {
-                frameworks.addAll(pod.frameworks);
                 if (platform.contains("ios")) {
-                    frameworks.addAll(pod.ios_frameworks);
+                    frameworks.addAll(pod.frameworks.ios);
                 }
                 else if (platform.contains("osx")) {
-                    frameworks.addAll(pod.osx_frameworks);
+                    frameworks.addAll(pod.frameworks.osx);
                 }
             }
             return new ArrayList<String>(frameworks);
@@ -93,15 +90,99 @@ public class CocoaPodsService {
         public List<String> getAllPodWeakFrameworks(String platform) {
             Set<String> weakFrameworks = new HashSet<>();
             for (PodSpec pod : pods) {
-                weakFrameworks.addAll(pod.weak_frameworks);
                 if (platform.contains("ios")) {
-                    weakFrameworks.addAll(pod.ios_weak_frameworks);
+                    weakFrameworks.addAll(pod.weak_frameworks.ios);
                 }
                 else if (platform.contains("osx")) {
-                    weakFrameworks.addAll(pod.osx_weak_frameworks);
+                    weakFrameworks.addAll(pod.weak_frameworks.osx);
                 }
             }
             return new ArrayList<String>(weakFrameworks);
+        }
+    }
+
+    public class LanguageSet {
+        public Set<String> c = new HashSet<>();
+        public Set<String> cpp = new HashSet<>();
+        public Set<String> objc = new HashSet<>();
+        public Set<String> objcpp = new HashSet<>();
+
+        public void add(String value) {
+            c.add(value);
+            cpp.add(value);
+            objc.add(value);
+            objcpp.add(value);
+        }
+
+        public void addAll(List<String> values) {
+            for (String v : values) {
+                add(v);
+            }
+        }
+        public void addAll(LanguageSet set) {
+            c.addAll(set.c);
+            cpp.addAll(set.cpp);
+            objc.addAll(set.objc);
+            objcpp.addAll(set.objcpp);
+        }
+
+        @Override
+        public String toString() {
+            StringBuilder sb = new StringBuilder();
+            sb.append("c: " + c);
+            sb.append("cpp: " + cpp);
+            sb.append("objc: " + objc);
+            sb.append("objcpp: " + objcpp);
+            return sb.toString();
+        }
+    }
+
+    public class PlatformAndLanguageSet {
+        public LanguageSet ios = new LanguageSet();
+        public LanguageSet osx = new LanguageSet();
+
+        public void addAll(PlatformAndLanguageSet v) {
+            ios.addAll(v.ios);
+            osx.addAll(v.osx);
+        }
+        public void addAll(List<String> values) {
+            for (String v :  values) {
+                ios.add(v);
+                osx.add(v);
+            }
+        }
+
+        @Override
+        public String toString() {
+            StringBuilder sb = new StringBuilder();
+            sb.append("ios:\n" + ios.toString());
+            sb.append("osx:\n" + osx.toString());
+            return sb.toString();
+        }
+    }
+
+    public class PlatformSet {
+        public Set<String> ios = new HashSet<>();
+        public Set<String> osx = new HashSet<>();
+
+        public void addAll(PlatformSet v) {
+            ios.addAll(v.ios);
+            osx.addAll(v.osx);
+        }
+
+        public void addAll(List<String> values) {
+            for (String v :  values) {
+                ios.add(v);
+                osx.add(v);
+            }
+        }
+
+        @Override
+        public String toString() {
+            StringBuilder sb = new StringBuilder();
+            sb.append("ios:\n" + ios.toString());
+            sb.append("osx:\n" + osx.toString());
+            return sb.toString();
         }
     }
 
@@ -115,25 +196,14 @@ public class CocoaPodsService {
         public Set<File> includePaths = new HashSet<>();
         public PodSpec parentSpec = null;
         public List<PodSpec> subspecs = new ArrayList<>();
-        public Set<String> defines = new HashSet<>();
-        public Set<String> ios_defines = new HashSet<>();
-        public Set<String> osx_defines = new HashSet<>();
-        public Set<String> flags = new HashSet<>();
-        public Set<String> ios_flags = new HashSet<>();
-        public Set<String> osx_flags = new HashSet<>();
-        public Set<String> linkflags = new HashSet<>();
-        public Set<String> ios_linkflags = new HashSet<>();
-        public Set<String> osx_linkflags = new HashSet<>();
+
+        public PlatformAndLanguageSet flags = new PlatformAndLanguageSet();
+        public PlatformSet defines = new PlatformSet();
+        public PlatformSet linkflags = new PlatformSet();
         public Set<String> vendoredframeworks = new HashSet<>();
-        public Set<String> weak_frameworks = new HashSet<>();
-        public Set<String> ios_weak_frameworks = new HashSet<>();
-        public Set<String> osx_weak_frameworks = new HashSet<>();
-        public Set<String> frameworks = new HashSet<>();
-        public Set<String> ios_frameworks = new HashSet<>();
-        public Set<String> osx_frameworks = new HashSet<>();
-        public Set<String> libraries = new HashSet<>();
-        public Set<String> ios_libraries = new HashSet<>();
-        public Set<String> osx_libraries = new HashSet<>();
+        public PlatformSet weak_frameworks = new PlatformSet();
+        public PlatformSet frameworks = new PlatformSet();
+        public PlatformSet libraries = new PlatformSet();
         public File dir;
 
         public String toString(String indentation) {
@@ -143,26 +213,12 @@ public class CocoaPodsService {
             sb.append(indentation + "  src: " + sourceFiles + "\n");
             sb.append(indentation + "  includes: " + includePaths + "\n");
             sb.append(indentation + "  defines: " + defines + "\n");
-            sb.append(indentation + "  ios_defines: " + ios_defines + "\n");
-            sb.append(indentation + "  osx_defines: " + osx_defines + "\n");
             sb.append(indentation + "  flags: " + flags + "\n");
-            sb.append(indentation + "  ios_flags: " + ios_flags + "\n");
-            sb.append(indentation + "  osx_flags: " + osx_flags + "\n");
             sb.append(indentation + "  linkflags: " + linkflags + "\n");
-            sb.append(indentation + "  ios_linkflags: " + ios_linkflags + "\n");
-            sb.append(indentation + "  osx_linkflags: " + osx_linkflags + "\n");
-            sb.append(indentation + "  iosversion: " + iosversion + "\n");
-            sb.append(indentation + "  osxversion: " + osxversion + "\n");
             sb.append(indentation + "  weak_frameworks: " + weak_frameworks + "\n");
-            sb.append(indentation + "  ios_weak_frameworks: " + ios_weak_frameworks + "\n");
-            sb.append(indentation + "  osx_weak_frameworks: " + osx_weak_frameworks + "\n");
             sb.append(indentation + "  frameworks: " + frameworks + "\n");
-            sb.append(indentation + "  ios_frameworks: " + ios_frameworks + "\n");
-            sb.append(indentation + "  osx_frameworks: " + osx_frameworks + "\n");
             sb.append(indentation + "  vendoredframeworks: " + vendoredframeworks + "\n");
             sb.append(indentation + "  libraries: " + libraries + "\n");
-            sb.append(indentation + "  ios_libraries: " + ios_libraries + "\n");
-            sb.append(indentation + "  osx_libraries: " + osx_libraries + "\n");
             sb.append(indentation + "  parentSpec: " + ((parentSpec != null) ? parentSpec.name : "null") + "\n");
             for (PodSpec sub : subspecs) {
                 sb.append(sub.toString(indentation + "  "));
@@ -451,7 +507,7 @@ public class CocoaPodsService {
         return value.equals(expected);
     }
 
-    private void parseConfig(JSONObject config, Set<String> flags, Set<String> linkflags, Set<String> defines) {
+    private void parseConfig(JSONObject config, LanguageSet flags, Set<String> linkflags, Set<String> defines) {
         // https://pewpewthespells.com/blog/buildsettings.html
         // defines
         if (hasString(config, "GCC_PREPROCESSOR_DEFINITIONS")) {
@@ -475,7 +531,8 @@ public class CocoaPodsService {
                 case "compiler-default": 
                 default:  compilerFlag = "-std=gnu++98"; break;
             }
-            flags.add(compilerFlag);
+            flags.cpp.add(compilerFlag);
+            flags.objcpp.add(compilerFlag);
         }
         if (hasString(config, "GCC_C_LANGUAGE_STANDARD")) {
             String cStandard = getAsString(config, "GCC_C_LANGUAGE_STANDARD", "compiler-default");
@@ -491,7 +548,7 @@ public class CocoaPodsService {
                 case "compiler-default": 
                 default:  compilerFlag = "-std=gnu99"; break;
             }
-            flags.add(compilerFlag);
+            flags.c.add(compilerFlag);
         }
         if (hasString(config, "CLANG_CXX_LIBRARY")) {
             String stdLib = getAsString(config, "CLANG_CXX_LIBRARY", "compiler-default");
@@ -502,13 +559,14 @@ public class CocoaPodsService {
                 case "compiler-default":
                 default: stdLibFlag = "-stdlib=libstdlibc++"; break;
             }
-            flags.add(stdLibFlag);
+            flags.cpp.add(stdLibFlag);
+            flags.objcpp.add(stdLibFlag);
         }
         if (compareString(config, "GCC_ENABLE_CPP_EXCEPTIONS", "YES")) {
-            flags.add("-fcxx-exceptions");
+            flags.cpp.add("-fcxx-exceptions");
         }
         if (compareString(config, "GCC_ENABLE_CPP_EXCEPTIONS", "NO")) {
-            flags.add("-fno-cxx-exceptions");
+            flags.cpp.add("-fno-cxx-exceptions");
         }
         if (compareString(config, "GCC_ENABLE_EXCEPTIONS", "YES")) {
             flags.add("-fexceptions");
@@ -517,22 +575,26 @@ public class CocoaPodsService {
             flags.add("-fno-exceptions");
         }
         if (compareString(config, "GCC_ENABLE_OBJC_EXCEPTIONS", "YES")) {
-            flags.add("-fobjc-exceptions");
+            flags.objc.add("-fobjc-exceptions");
+            flags.objcpp.add("-fobjc-exceptions");
         }
         if (compareString(config, "GCC_ENABLE_OBJC_EXCEPTIONS", "NO")) {
-            flags.add("-fno-objc-exceptions");
+            flags.objc.add("-fno-objc-exceptions");
+            flags.objcpp.add("-fno-objc-exceptions");
         }
         if (compareString(config, "GCC_ENABLE_CPP_RTTI", "YES")) {
-            flags.add("-frtti");
+            flags.cpp.add("-frtti");
         }
         if (compareString(config, "GCC_ENABLE_CPP_RTTI", "NO")) {
-            flags.add("-fno-rtti");
+            flags.cpp.add("-fno-rtti");
         }
         if (compareString(config, "GCC_ENABLE_OBJC_GC", "supported")) {
-            flags.add("-fobjc-gc");
+            flags.objc.add("-fobjc-gc");
+            flags.objcpp.add("-fobjc-gc");
         }
         if (compareString(config, "GCC_ENABLE_OBJC_GC", "required")) {
-            flags.add("-fobjc-gc-only");
+            flags.objc.add("-fobjc-gc-only");
+            flags.objcpp.add("-fobjc-gc-only");
         }
         if (compareString(config, "GCC_ENABLE_ASM_KEYWORD", "YES")) {
             flags.add("-fasm");
@@ -544,11 +606,12 @@ public class CocoaPodsService {
 
     private void parseMultiPlatformConfig(PodSpec spec, JSONObject config) {
         if (config != null) {
-            parseConfig(config, spec.flags, spec.linkflags, spec.defines);
+            parseConfig(config, spec.flags.ios, spec.linkflags.ios, spec.defines.ios);
+            parseConfig(config, spec.flags.osx, spec.linkflags.osx, spec.defines.osx);
             JSONObject iosConfig = (JSONObject)config.get("ios");
             JSONObject osxConfig = (JSONObject)config.get("ios");
-            if (iosConfig != null) parseConfig(iosConfig, spec.ios_flags, spec.ios_linkflags, spec.defines);
-            if (osxConfig != null) parseConfig(osxConfig, spec.osx_flags, spec.osx_linkflags, spec.defines);
+            if (iosConfig != null) parseConfig(iosConfig, spec.flags.ios, spec.linkflags.ios, spec.defines.ios);
+            if (osxConfig != null) parseConfig(osxConfig, spec.flags.osx, spec.linkflags.osx, spec.defines.osx);
         }
     }
 
@@ -562,11 +625,12 @@ public class CocoaPodsService {
 
         // inherit flags and defines from the parent
         if (parent != null) {
-            spec.flags.addAll(parent.flags);
-            spec.ios_flags.addAll(parent.ios_flags);
-            spec.osx_flags.addAll(parent.osx_flags);
-            spec.defines.addAll(parent.defines);
-            spec.linkflags.addAll(parent.linkflags);
+            spec.flags.ios.addAll(parent.flags.ios);
+            spec.flags.osx.addAll(parent.flags.osx);
+            spec.defines.ios.addAll(parent.defines.ios);
+            spec.defines.osx.addAll(parent.defines.osx);
+            spec.linkflags.ios.addAll(parent.linkflags.ios);
+            spec.linkflags.ios.addAll(parent.linkflags.ios);
         }
 
         // platform versions
@@ -594,22 +658,37 @@ public class CocoaPodsService {
         Boolean requiresArc = true;
         Object requiresArcObject = specJson.get("requires_arc");
         if (requiresArcObject instanceof Boolean) requiresArc = (Boolean)requiresArcObject;
-        spec.flags.add((requiresArc == null || requiresArc == true) ? "-fobjc-arc" : "-fno-objc-arc");
-        spec.flags.addAll(getAsSplitString(specJson, "compiler_flags"));
+        spec.flags.ios.objc.add((requiresArc == null || requiresArc == true) ? "-fobjc-arc" : "-fno-objc-arc");
+        spec.flags.ios.objcpp.add((requiresArc == null || requiresArc == true) ? "-fobjc-arc" : "-fno-objc-arc");
+        spec.flags.osx.objc.add((requiresArc == null || requiresArc == true) ? "-fobjc-arc" : "-fno-objc-arc");
+        spec.flags.osx.objcpp.add((requiresArc == null || requiresArc == true) ? "-fobjc-arc" : "-fno-objc-arc");
+        
+        // compiler flags
+        spec.flags.ios.addAll(getAsSplitString(specJson, "compiler_flags"));
+        spec.flags.osx.addAll(getAsSplitString(specJson, "compiler_flags"));
+
+        spec.flags.ios.c.add("--language=c");
+        spec.flags.osx.c.add("--language=c");
+        spec.flags.ios.cpp.add("--language=c++");
+        spec.flags.osx.cpp.add("--language=c++");
+        spec.flags.ios.objc.add("--language=objective-c");
+        spec.flags.osx.objc.add("--language=objective-c");
+        spec.flags.ios.objcpp.add("--language=objective-c++");
+        spec.flags.osx.objcpp.add("--language=objective-c++");
 
         // platform specific flags
-        if (ios != null) spec.ios_flags.addAll(getAsJSONArray(ios, "compiler_flags"));
-        if (osx != null) spec.osx_flags.addAll(getAsJSONArray(osx, "compiler_flags"));
+        if (ios != null) spec.flags.ios.addAll(getAsJSONArray(ios, "compiler_flags"));
+        if (osx != null) spec.flags.osx.addAll(getAsJSONArray(osx, "compiler_flags"));
 
         // frameworks
         spec.frameworks.addAll(getAsJSONArray(specJson, "frameworks"));
-        if (ios != null) spec.ios_frameworks.addAll(getAsJSONArray(ios, "frameworks"));
-        if (osx != null) spec.osx_frameworks.addAll(getAsJSONArray(osx, "frameworks"));
+        if (ios != null) spec.frameworks.ios.addAll(getAsJSONArray(ios, "frameworks"));
+        if (osx != null) spec.frameworks.osx.addAll(getAsJSONArray(osx, "frameworks"));
 
         // weak frameworks
         spec.weak_frameworks.addAll(getAsJSONArray(specJson, "weak_frameworks"));
-        if (ios != null) spec.ios_weak_frameworks.addAll(getAsJSONArray(ios, "weak_frameworks"));
-        if (osx != null) spec.osx_weak_frameworks.addAll(getAsJSONArray(osx, "weak_frameworks"));
+        if (ios != null) spec.weak_frameworks.ios.addAll(getAsJSONArray(ios, "weak_frameworks"));
+        if (osx != null) spec.weak_frameworks.osx.addAll(getAsJSONArray(osx, "weak_frameworks"));
 
         // vendored_frameworks
         JSONArray vendored = getAsJSONArray(specJson, "vendored_frameworks");
@@ -631,8 +710,8 @@ public class CocoaPodsService {
 
         // libraries
         spec.libraries.addAll(getAsJSONArray(specJson, "libraries"));
-        if (ios != null) spec.ios_libraries.addAll(getAsJSONArray(ios, "libraries"));
-        if (osx != null) spec.osx_libraries.addAll(getAsJSONArray(osx, "libraries"));
+        if (ios != null) spec.libraries.ios.addAll(getAsJSONArray(ios, "libraries"));
+        if (osx != null) spec.libraries.osx.addAll(getAsJSONArray(osx, "libraries"));
 
         // parse subspecs
         JSONArray subspecs = getAsJSONArray(specJson, "subspecs");

--- a/server/src/main/java/com/defold/extender/services/CocoaPodsService.java
+++ b/server/src/main/java/com/defold/extender/services/CocoaPodsService.java
@@ -478,7 +478,20 @@ public class CocoaPodsService {
             flags.add(compilerFlag);
         }
         if (hasString(config, "GCC_C_LANGUAGE_STANDARD")) {
-            flags.add("-std=" + getAsString(config, "GCC_C_LANGUAGE_STANDARD"));
+            String cStandard = getAsString(config, "GCC_C_LANGUAGE_STANDARD", "compiler-default");
+            String compilerFlag = "";
+            switch (cStandard) {
+                case "ansi":  compilerFlag = "-ansi"; break;
+                case "c89":   compilerFlag = "-std=c89"; break;
+                case "gnu89": compilerFlag = "std=gnu89"; break;
+                case "c99":   compilerFlag = "-std=c99"; break;
+                case "c11":   compilerFlag = "-std=c11"; break;
+                case "gnu11": compilerFlag = "-std=gnu11"; break;
+                case "gnu99":
+                case "compiler-default": 
+                default:  compilerFlag = "-std=gnu99"; break;
+            }
+            flags.add(compilerFlag);
         }
         if (hasString(config, "CLANG_CXX_LIBRARY")) {
             String stdLib = getAsString(config, "CLANG_CXX_LIBRARY", "compiler-default");

--- a/server/src/main/java/com/defold/extender/services/CocoaPodsService.java
+++ b/server/src/main/java/com/defold/extender/services/CocoaPodsService.java
@@ -32,7 +32,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.HashMap;
 import java.util.Set;
-import java.util.HashSet;
+import java.util.LinkedHashSet;
 import java.util.Iterator;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
@@ -49,7 +49,7 @@ public class CocoaPodsService {
         public String platformMinVersion;
 
         public List<String> getAllPodLibs(String platform) {
-            Set<String> libs = new HashSet<>();
+            Set<String> libs = new LinkedHashSet<>();
             for (PodSpec pod : pods) {
                 if (platform.contains("ios")) {
                     libs.addAll(pod.libraries.ios);
@@ -62,7 +62,7 @@ public class CocoaPodsService {
         }
 
         public List<String> getAllPodLinkFlags(String platform) {
-            Set<String> flags = new HashSet<>();
+            Set<String> flags = new LinkedHashSet<>();
             for (PodSpec pod : pods) {
                 if (platform.contains("ios")) {
                     flags.addAll(pod.linkflags.ios);
@@ -75,7 +75,7 @@ public class CocoaPodsService {
         }
 
         public List<String> getAllPodFrameworks(String platform) {
-            Set<String> frameworks = new HashSet<>();
+            Set<String> frameworks = new LinkedHashSet<>();
             for (PodSpec pod : pods) {
                 if (platform.contains("ios")) {
                     frameworks.addAll(pod.frameworks.ios);
@@ -88,7 +88,7 @@ public class CocoaPodsService {
         }
 
         public List<String> getAllPodWeakFrameworks(String platform) {
-            Set<String> weakFrameworks = new HashSet<>();
+            Set<String> weakFrameworks = new LinkedHashSet<>();
             for (PodSpec pod : pods) {
                 if (platform.contains("ios")) {
                     weakFrameworks.addAll(pod.weak_frameworks.ios);
@@ -102,10 +102,10 @@ public class CocoaPodsService {
     }
 
     public class LanguageSet {
-        public Set<String> c = new HashSet<>();
-        public Set<String> cpp = new HashSet<>();
-        public Set<String> objc = new HashSet<>();
-        public Set<String> objcpp = new HashSet<>();
+        public Set<String> c = new LinkedHashSet<>();
+        public Set<String> cpp = new LinkedHashSet<>();
+        public Set<String> objc = new LinkedHashSet<>();
+        public Set<String> objcpp = new LinkedHashSet<>();
 
         public void add(String value) {
             c.add(value);
@@ -162,8 +162,8 @@ public class CocoaPodsService {
     }
 
     public class PlatformSet {
-        public Set<String> ios = new HashSet<>();
-        public Set<String> osx = new HashSet<>();
+        public Set<String> ios = new LinkedHashSet<>();
+        public Set<String> osx = new LinkedHashSet<>();
 
         public void addAll(PlatformSet v) {
             ios.addAll(v.ios);
@@ -192,15 +192,15 @@ public class CocoaPodsService {
         public String originalJSON = "";
         public String iosversion = "";
         public String osxversion = "";
-        public Set<File> sourceFiles = new HashSet<>();
-        public Set<File> includePaths = new HashSet<>();
+        public Set<File> sourceFiles = new LinkedHashSet<>();
+        public Set<File> includePaths = new LinkedHashSet<>();
         public PodSpec parentSpec = null;
         public List<PodSpec> subspecs = new ArrayList<>();
 
         public PlatformAndLanguageSet flags = new PlatformAndLanguageSet();
         public PlatformSet defines = new PlatformSet();
         public PlatformSet linkflags = new PlatformSet();
-        public Set<String> vendoredframeworks = new HashSet<>();
+        public Set<String> vendoredframeworks = new LinkedHashSet<>();
         public PlatformSet weak_frameworks = new PlatformSet();
         public PlatformSet frameworks = new PlatformSet();
         public PlatformSet libraries = new PlatformSet();
@@ -371,7 +371,7 @@ public class CocoaPodsService {
      * @return List of include paths
      */
     private List<File> findPodIncludePaths(PodSpec pod, String pattern) {
-        Set<File> includePaths = new HashSet<>();
+        Set<File> includePaths = new LinkedHashSet<>();
 
         String absolutePatternPath = pod.dir.getAbsolutePath() + File.separator + pattern;
         List<File> podSrcFiles = ExtenderUtil.listFilesGlob(pod.dir, absolutePatternPath);
@@ -540,7 +540,7 @@ public class CocoaPodsService {
             switch (cStandard) {
                 case "ansi":  compilerFlag = "-ansi"; break;
                 case "c89":   compilerFlag = "-std=c89"; break;
-                case "gnu89": compilerFlag = "std=gnu89"; break;
+                case "gnu89": compilerFlag = "-std=gnu89"; break;
                 case "c99":   compilerFlag = "-std=c99"; break;
                 case "c11":   compilerFlag = "-std=c11"; break;
                 case "gnu11": compilerFlag = "-std=gnu11"; break;
@@ -787,7 +787,7 @@ public class CocoaPodsService {
             - GoogleAppMeasurement (= 8.13.0)
         */
         List<String> lines = Files.readAllLines(podFileLock.toPath());
-        Set<String> pods = new HashSet<>();
+        Set<String> pods = new LinkedHashSet<>();
         while (!lines.isEmpty()) if (lines.remove(0).startsWith("PODS:")) break;
         while (!lines.isEmpty()) {
             String line = lines.remove(0);

